relational_operators/HashJoinOperator.cpp

```diff
@@ -484,7 +565,38 @@ void HashInnerJoinWorkOrder::execute() {
   const relation_id build_relation_id = build_relation_.getID();
   const relation_id probe_relation_id = probe_relation_.getID();
 
-  for (std::pair<const block_id, std::pair<std::vector<tuple_id>, std::vector<tuple_id>>>
+  // Create a map of ValueAccessors and what attributes we want to pick from them.
+  std::vector<std::pair<ValueAccessor *, std::vector<attribute_id>>> accessor_attribute_map;
+  const std::size_t build_index = 0, probe_index = 1, temp_index = 2;
+  for (std::size_t i = 0; i < 3; ++i) {
```

Comment:
Please replace the magic number `3` with some constant like `kNumIndexes`.

Also please rename above three `indexes` to `kBuildIndex`, `kProbeIndex`, and `kTempIndex`, respectively.